neqo-transport/src/connection/mod.rs

```diff
@@ -917,15 +917,20 @@ impl Connection {
     }
 
     /// Process new input datagrams on the connection.
-    pub fn process_input(&mut self, d: Datagram, now: Instant) {
+    pub fn process_input(&mut self, d: &Datagram, now: Instant) {
         self.input(d, now, now);
         self.process_saved(now);
         self.streams.cleanup_closed_streams();
     }
 
     /// Process new input datagrams on the connection.
-    pub fn process_multiple_input(&mut self, dgrams: Vec<Datagram>, now: Instant) {
-        if dgrams.is_empty() {
+    pub fn process_multiple_input<'a>(
+        &mut self,
+        dgrams: impl IntoIterator<Item = &'a Datagram>,
```

Comment:
I don't have a problem with requiring ExactSizeIterator for this, especially to avoid having to wrap in a peekable.  See the previous comment.

Reply:
See 3f76accc.